pkg/secrets_reflector/secrets_reflector.go

```
@@ -0,0 +1,153 @@
+	defaultSecretStore = "default"
+	defaultTimeout     = 5 * time.Second
+)
+
+// SecretsReflector defines the interface exposed by this package.
+type SecretsReflector interface {
+	Reflect(namespace, name string) error
+}
+
+// DCOSSecretsAPI defines the interface required by the
+// SecretsReflector to manage DC/OS secrets.
+type DCOSSecretsClient interface {
+	CreateSecret(ctx context.Context, store string, pathToSecret string, secretsV1Secret dcos.SecretsV1Secret) (*http.Response, error)
+	DeleteSecret(ctx context.Context, store string, pathToSecret string) (*http.Response, error)
+}
+
+// secretsReflector implements SecretsReflector interface.
+type secretsReflector struct {
+	dcosSecretsClient     DCOSSecretsClient
+	kubeCache             dklbcache.KubernetesResourceCache
+	kubeClient            kubernetes.Interface
+	kubernetesClusterName string
+	logger                log.FieldLogger
+}
+
+// New returns an instance of SecretsReflector.
+func New(kubernetesClusterName string, dcosSecretsClient DCOSSecretsClient, kubeCache dklbcache.KubernetesResourceCache, kubeClient kubernetes.Interface) SecretsReflector {
+	return &secretsReflector{
+		dcosSecretsClient:     dcosSecretsClient,
+		kubeCache:             kubeCache,
+		kubeClient:            kubeClient,
+		kubernetesClusterName: kubernetesClusterName,
+		logger:                log.WithField("component", "secrets_reflector"),
+	}
+}
+
+// Reflect reads kubernetes secret with name from namespace,
+// translates it to a DCOS secret and checks to if it needs to
+// re-create it DC/OS.
+func (s *secretsReflector) Reflect(namespace, name string) error {
+	// get secret from kubernetes
+	kubeSecret, err := s.kubeCache.GetSecret(namespace, name)
+	if err != nil {
+		s.logger.Errorf("failed to get secret %s/%s", namespace, name)
+		return err
+	}
+	// translate kubernetes secret to a dcos secret
+	dcosSecret, err := s.translate(kubeSecret)
+	if err != nil {
+		s.logger.Errorf("failed to translate secret: %s", err)
```

```suggestion
		s.logger.Errorf("failed to translate secret \"%s/%s\": %v", namespace, name, err)
```

---

Again, can we return the error to be logged higher up as suggested above by @bmcstdio?